### DISCUSSION TITLE
Gitlab project access token

### DISCUSSION THREAD

**Discussion by hieroshima** [renovatebot/renovate#9099] [Request Help]
Hi,

I would like to ask if there is a possibility to use Renovate Bot in GitLab with authenticating for each project separately, by using the project access token?

I have a situation that I don't have any extra user to use as a Renovate Bot. Now, I'm using my user personal token, but all MRs are created as Me. I would like to avoid it. From GitLab documentation, I know that the deploy key and deploy token is not supported for GitLab API - so the only way to use API is a personal token (which I currently use) and project access token (which I would like to use).

Unfortunately creating a new user for bot is not possible...

Thanks for any suggestions

**Comment by rarkins** [comment]
We currently assume one token per run, meaning all repositories in a run need to share a token.  If it were a project token then it would need to run once per project (which is actually not as bad as it sounds) however you'd need to write the wrapper which:

- works out the list of projects (statically or through an api call with personal token)
- iterate through the list, generating a project token and passing it to renovate 

Generating tokens each run seems non optimal but I don't know how else to store them

**Reply by viceice** [reply]
I think those token are fixed, gitlab creates a internal user which only project access. So the token is essentially a PAT scoped to a repo.

So we would need to create those token for each repo and then let renovate run in a loop over each repo / token. Alternatively use multiple shcedules with different environment passing token and repo
